iot-hub/Samples/service/CleanUpDevicesSample/CleanUpDevicesSample.cs

```diff
@@ -1,113 +1,149 @@
-                            {
-                                _devicesToDelete.Add(new Device(deviceId));
-                            }
-                        }
-                    }
-                }
-
-                var _bulkDeleteList = new List<Device>(DeleteBatchSize);
-                while (_devicesToDelete.Count > 0)
-                {
-                    int i;
-                    for (i = 0; (i < DeleteBatchSize) && (i < _devicesToDelete.Count); i++)
-                    {
-                        _bulkDeleteList.Add(_devicesToDelete[i]);
-                        Console.WriteLine($"\tRemove: {_devicesToDelete[i].Id}");
-                    }
-
-                    _devicesToDelete.RemoveRange(0, i);
-
-                    BulkRegistryOperationResult ret = await _rm.RemoveDevices2Async(_bulkDeleteList, true, CancellationToken.None).ConfigureAwait(false);
-                    devicesDeleted += _bulkDeleteList.Count - ret.Errors.Length;
-                    Console.WriteLine($"BATCH DELETE: {devicesDeleted}/{_bulkDeleteList.Count}");
-                    if (!ret.IsSuccessful)
-                    {
-                        foreach (DeviceRegistryOperationError error in ret.Errors)
-                        {
-                            Console.WriteLine($"\tERROR: {error.DeviceId} - {error.ErrorCode}: {error.ErrorStatus}");
-                        }
-                    }
-
-                    _bulkDeleteList.Clear();
-                }
-
-                Console.WriteLine($"-- Total no of devices deleted: {devicesDeleted}");
-            }
-            catch (Exception ex)
-            {
-                Console.WriteLine(ex);
-            }
-        }
-
-        private async Task PrintDeviceCount()
-        {
-            string countSqlQuery = "SELECT COUNT() AS numberOfDevices FROM devices";
-            IQuery countQuery = _rm.CreateQuery(countSqlQuery);
-            while (countQuery.HasMoreResults)
-            {
-                IEnumerable<string> result = await countQuery.GetNextAsJsonAsync().ConfigureAwait(false);
-                foreach (var item in result)
-                {
-                    Console.WriteLine($"Total no of devices on the hub: \n{item}");
-                }
-            }
-        }
-    }
-}
-
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.Devices.Samples
+{
+    public class CleanUpDevicesSample
+    {
+        private const int QueryBatchSize = 1000;
+        private const int DeleteBatchSize = 100;
+        private RegistryManager _rm;
+        private List<string> _deleteDeviceWithPrefix =
+            new List<string>{
+                // C# E2E tests
+                "E2E_",
```

Comment:
What E2E are we talking about in sample?

Reply:
This sample is used to cleanup the Azure resources used for our E2E tests